packages/application-shell/src/components/custom-view-authenticated-shell/custom-view-authenticated-shell.tsx

```diff
@@ -0,0 +1,100 @@
+import { type ReactNode } from 'react';
+import { PageUnauthorized } from '@commercetools-frontend/application-components';
+import { ApplicationContextProvider } from '@commercetools-frontend/application-shell-connectors';
+import { ApplicationWindow } from '@commercetools-frontend/constants';
+import {
+  AsyncLocaleData,
+  type TAsyncLocaleDataProps,
+} from '@commercetools-frontend/i18n';
+import { ThemeProvider } from '@commercetools-uikit/design-system';
+import ApplicationLoader from '../application-loader';
+import { getBrowserLocale } from '../application-shell-provider/utils';
+import ConfigureIntlProvider from '../configure-intl-provider';
+import FetchProject from '../fetch-project';
+import FetchUser from '../fetch-user';
+import SetupFlopFlipProvider from '../setup-flop-flip-provider';
+
+type TCustomViewAuthenticatedShellProps = {
+  dataLocale: string;
+  environment: ApplicationWindow['app'];
+  messages: TAsyncLocaleDataProps['applicationMessages'];
+  projectKey?: string;
+  children: ReactNode;
+};
+
+function CustomViewAuthenticatedShell(
+  props: TCustomViewAuthenticatedShellProps
+) {
+  const browserLocale = getBrowserLocale(window);
+
+  return (
+    <FetchUser>
+      {({ isLoading, error: fetchUserError, user }) => {
+        if (isLoading) {
+          return <ApplicationLoader />;
+        }
+
+        return (
+          <AsyncLocaleData
+            locale={user?.language || browserLocale}
+            applicationMessages={props.messages}
+          >
+            {({ isLoading: isLoadingLocaleData, locale, messages }) => {
+              return (
+                <ConfigureIntlProvider
+                  // We do not want to pass the language as long as the locale data
+                  // is not loaded.
+                  {...(isLoadingLocaleData ? {} : { locale, messages })}
+                >
+                  <ApplicationContextProvider
+                    user={user}
+                    environment={props.environment}
+                  >
+                    <SetupFlopFlipProvider
+                      user={user}
+                      projectKey={props.projectKey}
+                      ldClientSideId={props.environment.ldClientSideId}
+                      // flags={props.featureFlags}
+                      // defaultFlags={props.defaultFeatureFlags}
```

Comment:
Not sure if we want to allow feature flags in Custom Views. If so, also not sure which ones should we pass (defaultFeatureFlags?).

Reply:
Yeah good point, we can omit the provider for now. Also (external) Custom Applications can't use the feature flags, it's only for internal usage.